freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/FreeIpaClientFactory.java

```diff
@@ -129,6 +132,8 @@ private FreeIpaClient getFreeIpaClient(Stack stack, boolean withPing, boolean fo
                     }
                 }
                 return client.orElseThrow(() -> createFreeIpaUnableToBuildClient(new FreeIpaHostNotAvailableException("No FreeIPA client was available")));
+            } catch (RetryableFreeIpaClientException e) {
```

Comment:
I think the only place we can have issues during building the client is the connect part where we do the login. There is a logic there which would build a retryable exception based on status code. 
https://github.com/hortonworks/cloudbreak/blob/8b19e1905d7c2ee05478a131363f9c61f7cce468/freeipa/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClientBuilder.java#L232-L236
Do you think this would not work well for cluster proxy based connections? I can't remember if I tested this scenario last week, but I will (again) if you think this might be not handled here.

Reply:
Thanks.

I think this will work for cluster proxy/CCM errors and some direct errors during the connections.